libbeat/metric/system/process/helpers.go

```diff
@@ -79,10 +79,14 @@ func GetProcCPUPercentage(s0, s1 ProcState) ProcState {
 	}
 
 	timeDelta := s1.SampleTime.Sub(s0.SampleTime)
-	timeDeltaMillis := timeDelta / time.Millisecond
+	timeDeltaMillis := float64(timeDelta / time.Millisecond)
 	totalCPUDeltaMillis := int64(s1.CPU.Total.Ticks.ValueOr(0) - s0.CPU.Total.Ticks.ValueOr(0))
 
-	pct := float64(totalCPUDeltaMillis) / float64(timeDeltaMillis)
+	// In theory this can only happen if the time delta is 0, which is unlikely but possible.
+	if timeDeltaMillis == 0 {
```

Comment:
You are doing this comparison against a float now, which leaves it open to floating point precisions issues comparing against an exact value.